tests/BLUEPRINT/test_blueprint_cli.py

```diff
@@ -106,425 +100,461 @@ def wrapper_grouped_decorator(*args, **kwargs):
+    @mock_mode
+    def test_cli_force_rerun(
+        self,
+        mock_rclass,
+        mock_build,
+        mock_plot_xz,
+        mock_plot_xy,
+        mock_save_CAD_model,
+        rerun_flag,
+        tempdir,
+    ):
+        """
+        Test that bluemira can be rerun from the CLI when the force flag is on.
+        """
+        runner = CliRunner()
+
+        # Set flags and run bluemira cli.
+        default_flags = ["-i", INDIR, "-ri", REACTORNAME, "-o", tempdir]
+        flags = default_flags + rerun_flag
+        runner.invoke(cli, default_flags)  # First run
+        result = runner.invoke(cli, flags)  # Rerun with force flag on
+        assert result.exit_code == 0, traceback.print_exception(*result.exc_info)
+
+    @mock_mode
+    def test_cli_avoid_rerun(
+        self,
+        mock_rclass,
+        mock_build,
+        mock_plot_xz,
+        mock_plot_xy,
+        mock_save_CAD_model,
+        tempdir,
+    ):
+        """
+        Test that bluemira can not be rerun from the CLI when the force flag is off.
+        """
+        runner = CliRunner()
+
+        # Set flags and run bluemira cli.
+        flags = ["-i", INDIR, "-ri", REACTORNAME, "-o", tempdir]
+        runner.invoke(cli, flags)  # First run
+        result = runner.invoke(cli, flags)  # Rerun with force flag off
+        assert result.exit_code == 1, traceback.print_exception(*result.exc_info)
+
+    @mock_mode
+    def test_cli_invalid_flag(
+        self,
+        mock_rclass,
+        mock_build,
+        mock_plot_xz,
+        mock_plot_xy,
+        mock_save_CAD_model,
+        tempdir,
+    ):
+        """
+        Test that the bluemira CLI fails correctly when an invalid flag is passed.
+        """
+        invalid_flag = "--this_flag_does_not_exist"
+        assert_invoke(
+            cli,
+            INDIR,
+            REACTORNAME,
+            tempdir,
+            extra_flags=[invalid_flag],
+            expected_exit_code=2,
+        )
 
+    def test_cli_invalid_inputs(self, tempdir):
+        """
+        Test that the bluemira CLI fails correctly when invalid inputs are passed.
+        """
+        invalid_inputs = [
+            "this_file_does_not_exist.json",
+            "this_file_does_not_exist.json",
+            "this_file_does_not_exist.json",
+            "this_file_does_not_exist.json",
+        ]
+        assert_invoke(
+            cli,
+            INDIR,
+            REACTORNAME,
+            tempdir,
+            extra_flags=invalid_inputs,
+            expected_exit_code=1,
+        )
 
-def test_cli_invalid_inputs(tempdir):
-    """
-    Test that the BLUEPRINT CLI fails correctly when invalid inputs are passed.
-    """
-    runner = CliRunner()
+    @mock_mode
+    def test_datadir(
+        self,
+        mock_rclass,
+        mock_build,
+        mock_plot_xz,
+        mock_plot_xy,
+        mock_save_CAD_model,
+        tempdir,
+    ):
+        runner = CliRunner()
```

Comment:
This isnt used (sonarcloud not me).

Reply:
Should be sorted in the latest push (I wonder if there's a way to configure sonarcloud to exclude patterns in its regex for variable / method names).

Reply:
Maybe somehow we can parse the flake8 config because it seems like they're the same name issues as flake has